src/main/io/gps.c

```diff
@@ -1548,6 +1548,7 @@ static bool UBLOX_parse_gps(void)
         gpsSol.llh.lon = _buffer.posllh.longitude;
         gpsSol.llh.lat = _buffer.posllh.latitude;
         gpsSol.llh.altCm = _buffer.posllh.altitudeMslMm / 10;  //alt in cm
+        next_fix = (_buffer.status.fix_status & NAV_STATUS_FIX_VALID) && (_buffer.status.fix_type == FIX_3D);
```

Comment:
I think that we need advice here from GPS hardware experts.
At this point, the code may not have received a GPS status packet.
So I think the intent here is to accept the current status of next_fix, which in the initial run is false.
Only in the subsequent `MSG_STATUS` case do we receive 3D fix status info.  When this case first runs, it will set the `next fix` accordingly.
With my limited understanding of the code, we should only modify `next_fix` in the `MSG_STATUS` field, but we can update the `gpsSetFixState` here if that serves some purpose (eg setting it to off if we haven't yet received a status field).

Reply:
Fairly sure this change shouldn't be implemented

Reply:
Thanks. Updated.